### PR TITLE
[MIG][12.0] mgmtsystem_action

### PR DESCRIPTION
**Pull request by simahawk**
History sucks because all the modules are tied together because no atomic commits were made in the past. We have to migrate all the modules so if we squash history every time is going to be a big hell to handle because of too many conflicts.
See https://github.com/OCA/management-system/pull/244#issuecomment-498952002

### CODE REVIEW

File: mgmtsystem_action/demo/mgmtsystem_action.xml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
```
Comment (i-vyshnevska) [inline mgmtsystem_action/demo/mgmtsystem_action.xml:2]:
```suggestion
<odoo noupdate="1">
```

File: mgmtsystem_action/demo/mgmtsystem_action.xml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  <data noupdate="1">
```
Comment (i-vyshnevska) [inline mgmtsystem_action/demo/mgmtsystem_action.xml:3]:
```suggestion
```

File: mgmtsystem_action/demo/mgmtsystem_action.xml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+    </record>
+
+    <record id="demo_improvement" model="mgmtsystem.action">
+      <field name="name">Manage your quality management system with OpenERP :o)</field>
+      <field name="type_action">improvement</field>
+
+    </record>
+
+  </data>
```
Comment (i-vyshnevska) [inline mgmtsystem_action/demo/mgmtsystem_action.xml:30]:
```suggestion
```

File: mgmtsystem_action/models/mgmtsystem_action.py [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+    @api.model
+    def _stage_groups(self, stages=None, domain=None, order=None):
+        return self.env['mgmtsystem.action.stage'].search([], order=order)
+
+    @api.model
+    def create(self, vals):
+        if vals.get('reference', _('New')) == _('New'):
+            Sequence = self.env['ir.sequence']
+            vals['reference'] = Sequence.next_by_code('mgmtsystem.action')
+        action = super(MgmtsystemAction, self).create(vals)
```
Comment (i-vyshnevska) [inline mgmtsystem_action/models/mgmtsystem_action.py:120]:
```suggestion
        action = super().create(vals)
```

**Review by i-vyshnevska — APPROVED** [review]
LG, just nitpicking

**Review by max3903 — APPROVED** [review]

**Pull request merged**